### PR TITLE
suggested solution to improve searching complexity to O(1)

### PR DESCRIPTION
**Pull request by OlfaBensoussia**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -26,29 +26,31 @@
...
     def get(self, id):
-        for todo in self.todos:
-            # TODO : Improve the searching complexity to O(1) using hashmap structure
-            if todo['id'] == id:
-                return todo
+        Todo = {}
+        Todo['id'] = id
+        Todo['task'] = self.todos[id]
+        return Todo
         api.abort(404, "Todo {} doesn't exist".format(id))
```
Comment (yurilaaziz) [inline server.py:38]: in case of absent id ?

File: server.py [hunk 2]
```diff
@@ -26,29 +26,31 @@
...
+        todo['id'] = id = self.counter = self.counter + 1
+        todo['task'] = self.todos[id] = data['task']
         return todo
 
     def update(self, id, data):
-        todo = self.get(id)
-        todo.update(data)
+        todo = data
+        todo['id'] = id
+        todo['task'] = self.todos[id] = data['task']
```
Comment (yurilaaziz) [inline server.py:49]: `self.todos[id] = data['task'] ` is enough, you set todo to data already

File: server.py [hunk 3]
```diff
@@ -61,7 +63,7 @@ def delete(self, id):
 class TodoList(Resource):
     '''Shows a list of all todos, and lets you POST to add new tasks'''
     @ns.doc('list_todos')
-    @ns.marshal_list_with(todo)
+    # @ns.marshal_list_with(todo)
```
Comment (yurilaaziz) [inline server.py:66]: please keep marshaling

File: test_server.py [hunk 4]
```diff
@@ -11,11 +12,25 @@ def test_list_todo():
...
 
 
-def test_empty_todo():
+# TODO : add test raise exception when task is absent
+# If the task is empty
+def empty_todo():
     DAO = TodoDAO()
-    assert len(DAO.todos) == 0
+    empty = DAO.create({'task': ''})
+    if not empty['task']:
```
Comment (yurilaaziz) [inline test_server.py:20]: This should catch the exception raised by the class method.
Tests should never raise exceptions

File: test_server.py [hunk 5]
```diff
@@ -11,11 +12,25 @@ def test_list_todo():
...
+    DAO = TodoDAO()
+    none = DAO.create({'task': None})
+    if none['task'] is None:
+        raise Exception("Task not found")
 
 
-# TODO : add test raise exception when task is absent
 def test_notfound_todo():
-    pass
+    with pytest.raises(Exception):
```
Comment (yurilaaziz) [inline test_server.py:33]: similar to https://github.com/eyacom/fantastic-api/pull/10
unit tests code should check-raising exception from application code.

**Comment by OlfaBensoussia** [comment]
@yurilaaziz thank you for reviewing my PRs. 
Can I close this PR and make a new one with the requested changes ? After viewing the contribution guidelines, I realised that I need to put my changes in a new branch and I believe I can't do that in this PR and the other one.

**Pull request closed without merging**